resources/smpp/ra/src/main/java/org/restcomm/slee/resource/smpp/SmppSessionsImpl.java

```diff
@@ -495,4 +492,113 @@ public void fireUnknownThrowable(Throwable throwable) {
 
 	}
 
+	public class SenderThread extends Thread {
+
+		private Boolean running = true;
+		private LinkedBlockingQueue<SmppSendingTask> queue = new LinkedBlockingQueue<>();
+
+		public SenderThread(String name) {
+			super(name);
+		}
+
+		public void deactivate() {
+			running = false;
```

Comment:
seems we are mixing "custom flag stopping" with "standard thread interruption". I dont personally like using thread interruption as it has no business meaning, and it will cause undeterministic branches during "run" metho that will need to be handled per invocation...

I suggests to stick to "custom flag" and replace queue.take by queue.poll(timeout) with proper configurable timeout.

If we go on using "thread interruption", then lets remove runnign flag and use thread.isinterrupted instead.